persistence/scaladsl/src/main/scala/com/lightbend/lagom/scaladsl/persistence/ChildPersistentEntity.scala

```diff
@@ -0,0 +1,154 @@
+   * @param actorName The actor name.
+   */
+  def apply[P <: PersistentEntity: ClassTag](factory: () => P, entityId: String, actorName: String)(implicit ctx: ActorContext): ChildPersistentEntity[P#Command] = {
+
+    ctx.child(actorName) match {
+
+      case Some(actorRef) => new ChildPersistentEntity(actorRef)
+      case None =>
+
+        val conf = ctx.system.settings.config.getConfig("lagom.persistence")
+        val snapshotAfter: Option[Int] = conf.getString("snapshot-after") match {
+          case "off" => None
+          case _     => Some(conf.getInt("snapshot-after"))
+        }
+
+        val entity = factory()
+
+        val actorRef = ctx.actorOf(PersistentEntityActor.props(
+          persistenceIdPrefix = entity.entityTypeName,
+          entityId = Some(entityId),
+          entityFactory = () => entity,
+          snapshotAfter = snapshotAfter,
+          passivateAfterIdleTimeout = Duration.Undefined
+        ), actorName)
+
+        new ChildPersistentEntity(actorRef)
+    }
+  }
+}
+
+/**
+ * Factory for creating child persistent entities.
+ *
+ * This can be used instead of directly invoking [[ChildPersistentEntity.apply()]] as a method of indirection to
+ * allow you to inject a mocked actor (eg an Akka TestKit probe) when testing.
+ */
+trait ChildPersistentEntityFactory[P <: PersistentEntity] {
+
+  /**
+   * Create an entity.
+   *
+   * @param entityId  The id of the entity.
+   * @param actorName The name of the actor.
+   */
+  def apply(entityId: String, actorName: String)(implicit ctx: ActorContext): ChildPersistentEntity[P#Command]
+}
+
+object ChildPersistentEntityFactory {
+
+  /**
+   * Create a [[ChildPersistentEntityFactory]] for the given entity.
+   *
+   * @param factory The factory to create the entity.
+   */
+  def forEntity[P <: PersistentEntity: ClassTag](factory: () => P): ChildPersistentEntityFactory[P] = {
+    new ChildPersistentEntityFactory[P] {
+      override def apply(entityId: String, actorName: String)(implicit ctx: ActorContext): ChildPersistentEntity[P#Command] = {
+        ChildPersistentEntity(factory, entityId, actorName)
+      }
+    }
+  }
+
+  /**
+   * Create a mocked [[ChildPersistentEntityFactory]].
+   *
+   * All commands sent to any entities created will be passed as is to the `testProbe` actor. When `stop` is invoked,
+   * the actor will be sent a `PoisonPill`.
+   *
+   * This is intended to be used with an Akka TestKit TestProbe, for example:
+   *
+   * {{{
+   * val entityProbe = TestProbe("persistent-entity")
+   * val childEntityFactory = ChildPersistentEntityFactory.mocked[MyEntity](entityProbe.ref)
+   *
+   * // Create actor and cause it to send a command to the entity
+   * ...
+   *
+   * entityProbe.expectMsg(5.seconds, MyCommand)
+   * entityProbe.reply(SomeReply)
+   * }}}
+   *
+   * @param testProbe The test probe ActorRef.
+   */
+  def mocked[P <: PersistentEntity](testProbe: ActorRef): ChildPersistentEntityFactory[P] = {
```

Comment:
Reviewing the docs I noticed this.

Shouldn't `mocked` be moved out from production code and into `testkit`?

Reply:
Things in testkit are less discoverable. Perhaps mocked is a bad word here, perhaps `wrapped` would be a better word, ie this is an implementation of a factory that simply wraps an existing actor, rather than creates new actors. Personally, I'm not a big fan of separating utilities intended for testing from the production code, as it creates an artificial distinction that often encourages a hierarchy of code. Test code is just as important, if not more important, than production code, and so implementations that are more aimed at testing deserve to be alongside production code. There's of course competing concerns here - for example, if a test utility brings in other dependencies, then that's going to unnecessarily bloat a production artifact. But for something like this, I don't think such concerns exist, it's just an additional method that creates an additional very small class artifact.